treeherder/log_parser/tasks.py

```diff
@@ -19,85 +19,82 @@ def parser_task(f):
     """Decorator that ensures that log parsing task has not already run,
     and also adds New Relic annotations.
     """
-    def inner(job_log_id, priority):
-        newrelic.agent.add_custom_parameter("job_log_id", job_log_id)
-        job_log = JobLog.objects.select_related("job").get(id=job_log_id)
+    def inner(job_log):
+        logger.debug("parser_task for %s" % job_log.id)
+        newrelic.agent.add_custom_parameter("job_log_id", job_log.id)
         newrelic.agent.add_custom_parameter("job_log_name", job_log.name)
         newrelic.agent.add_custom_parameter("job_log_url", job_log.url)
         newrelic.agent.add_custom_parameter("job_log_status_prior",
                                             job_log.get_status_display())
         if job_log.status == JobLog.PARSED:
-            logger.info("log already parsed")
+            logger.info("%s log already parsed" % job_log.id)
             return True
 
-        return f(job_log, priority)
+        return f(job_log)
 
     inner.__name__ = f.__name__
     inner.__doc__ = f.__doc__
 
     return inner
 
 
-def parse_job_log(func_name, routing_key, job_log):
-    """
-    Schedule the log-related tasks to parse an individual log
-    """
-    task_funcs = {
-        "store_failure_lines": store_failure_lines,
-        "parse_log": parse_log,
+@retryable_task(name='log-parser', max_retries=10)
+def parse_logs(job_id, job_log_ids, priority):
+    job = Job.objects.get(id=job_id)
+    job_logs = JobLog.objects.filter(id__in=job_log_ids,
+                                     job=job)
+
+    if len(job_log_ids) != len(job_logs):
+        logger.warning("Failed to load all expected job ids: %s" % ", ".join(job_log_ids))
+    parser_tasks = {
+        "errorsummary_json": store_failure_lines,
+        "buildbot_text": parse_unstructured_log,
+        "builds-4h": parse_unstructured_log
     }
 
-    logger.debug("parse_job_log for job log %s (%s, %s)",
-                 job_log.id, func_name, routing_key)
-    priority = routing_key.rsplit(".", 1)[1]
-
-    signature = task_funcs[func_name].si(job_log.id, priority)
-    signature.set(routing_key=routing_key)
-
-    signature.apply_async()
+    seen_names = set()
+    for job_log in job_logs:
```

Comment:
Perhaps the "has this log already been parsed" check should be moved out of `@parser_tasks` to here?

Also, what's the intended behaviour for retries and failures for log parsing? eg should we retry X times then mark failed? Or mark failed straight away? Should we try to skip parsing one log (that fails) so we can at least parse the others?